### PR TITLE
XMLElementIter update for Julia 0.7

### PR DESCRIPTION
**Pull request by samtkaplan**
Hello,

Thanks for your great work on LightXML.  This change updates `XMLElementIter` for the new iteration protocal in Julia 0.7.  Without this change, I get a deprecation warning.

Thanks!

Sam

edit: cross-ref #87 which also has an iteration protocol update change in it for `XMLAttrIter`

### CODE REVIEW

File: src/nodes.jl [hunk 1]
```diff
@@ -259,11 +259,20 @@ struct XMLElementIter
...
-    ccall((:xmlFirstElementChild,libxml2), Xptr, (Xptr,), it.parent_ptr)
-Base.done(it::XMLElementIter, p::Xptr) = (p == C_NULL)
-Base.next(it::XMLElementIter, p::Xptr) =
-    (XMLElement(p), ccall((:xmlNextElementSibling,libxml2), Xptr, (Xptr,), p))
+if isdefined(Base, :iterate)
+    function Base.iterate(it::LightXML.XMLElementIter,
+            state=(ccall((:xmlFirstElementChild,LightXML.libxml2), LightXML.Xptr, (LightXML.Xptr,), it.parent_ptr), 0))
+        state[1] == C_NULL && return nothing
+        el = LightXML.XMLElement(state[1])
+        (el, (ccall((:xmlNextElementSibling,LightXML.libxml2), LightXML.Xptr, (LightXML.Xptr,), state[1]), 0))
```
Comment (tkelman) [inline src/nodes.jl]: don't need `LightXML.` qualification anywhere here, we're inside the package module

Comment (samtkaplan) [reply src/nodes.jl]: Nice Catch.  Thanks!  I removed the not-needed qualifications.

**Comment by tkoolen** [comment]
Ah sorry, I missed that one (and another one, actually). #87 is now updated to include those changes. Didn't see your PR before updating mine, otherwise I'd have rebased your commit onto the branch for #87.

**Comment by samtkaplan** [comment]
closing in favor of #87

**Pull request closed without merging**